app.rb

```diff
@@ -43,6 +43,19 @@
 end
 
 #
+#  Fetches a random definition by following the /random.php redirect
+#  and taking out the term with a regexp
+#
+get '/random' do
+  rdef = ud.get_random_definition()
+    unless valid_response? definition
+    raise NoDefinition, params[:term]
+  end
+
+  jsonp definition
```

Comment:
Here I tried to copy what you had, it's most likely broken, but I'll gladly take an explanation about what I should've written